api/room/schema.py

```diff
@@ -38,6 +40,58 @@ def save_room_tags(room, room_tags):
     room.save()
 
 
+class RoomResource(SQLAlchemyObjectType):
+    class Meta:
+        model = RoomResourceModel
+
+
+class AssignResource(graphene.Mutation):
+    """
+        Assigns a resource to a room
+    """
+    class Arguments:
+        resource_id = graphene.Int(required=True)
+        quantity = graphene.Int(required=True)
+        room_id = graphene.Int(required=True)
+    room_resource = graphene.Field(RoomResource)
+
+    @Auth.user_roles('Admin')
+    def mutate(self, info, **kwargs):
+        validate_empty_fields(**kwargs)
+        query = Room.get_query(info)
+        active_room = query.filter(RoomModel.state == "active")
+        exact_room = active_room.filter(
+            RoomModel.id == kwargs['room_id']).first()
+        if not exact_room:
+            raise GraphQLError("Room not found")
+        exact_resource = ResourceModel.query.filter_by(
+            id=kwargs['resource_id']).first()
+        if not exact_resource:
+            raise GraphQLError('invalid resource id')
+        exact_quantity = exact_resource.quantity
+        assigned_quantity = kwargs['quantity']
+        if assigned_quantity > exact_quantity:
+            raise GraphQLError(
+                'assigned resource cannot exceed quantity in database.'
+            )
+        if assigned_quantity < 1:
+            raise GraphQLError(
+                "assigned quantity cannot be less than 1."
+            )
+        exact_resource.quantity = exact_quantity - assigned_quantity
+        resource_query = RoomResource.get_query(info)
+        exact_resource_id = resource_query.filter(
```

Comment:
hey @d-kahara, Great work. Can you please add context on what this filter is doing?

Reply:
Hi @RachaelNantale , we're checking whether the resource id that is passed in as an argument already exists in the given room. If it does, we tell the user to update the quantity.